Riot/Modules/Common/Recents/RecentsViewController.m

```diff
@@ -2177,6 +2186,25 @@ - (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar
     [self.recentsSearchBar setShowsCancelButton:NO animated:NO];
 }
 
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [super searchBar:searchBar textDidChange:searchText];
+
+    UIImage *filterIcon;
+    if ([searchText length] == 0)
+    {
+        filterIcon = [UIImage imageNamed:@"filter_off"];
+    }
+    else
+    {
+        filterIcon = [UIImage imageNamed:@"threads_filter"];
```

Comment:
I'm not sure it's a good idea to reuse an icon from a completely separate feature here. Perhaps you should add an `filter_on` image or rename the threads one.

Reply:
On another note, I would've probably written this like so:
`UIImage *filterIcon = [UIImage imageNamed:(searchText.length > 0 ? @"filter_on" : @"filter_off")];`